### PR TITLE
feat: run jobs on every tf apply

### LINKED ISSUE #44: Address persistent GCE instance
`postdeployment.tf` uses a compute engine instance's metadata startup script to invoke the Cloud Run jobs created in `jobs.tf`. 

This instance currently keeps running throughout the life of the deployment, and has no purpose. It would be optimal to shutdown this instance after use to entire a lower ongoing cost.

Additionally, metadata startup scripts only fire on instance boot. Re-applying TF where an instance is running won't re-fire this script. 

The `client` job can be re-run without issue, but the `setup` job isn't currently idempotent, but can be adjusted. 

It would be optimal to ensure re-applying TF starts this instance up again, ensuring jobs are re-run to reassert state.

### PR DESCRIPTION
**Pull request by glasnt**
Requires https://github.com/GoogleCloudPlatform/avocano/pull/164 to be merged, released, and the new version referenced 

Resolves #44 

Cloud Run jobs are run through the metadata startup script on the GCE instance. Previously, when this terraform was first applied, this gce instance would stay on, and subsequent `terraform apply` commands would only alter the state of the job or gce instance, but would not _restart_ the instance, so the job execution would not be performed again.

With this change, after the gce instance has completed it's tasks, it shuts itself down. The TF designed state is running, so any time terraform is run, the instance will be restarted, and the jobs re-run. The change in avocano now ensures all these jobs are idempotent (previously, re-running the setup job would create duplicate products). 

This change now reasserts server image, client deploy, and database migrations. 

(Note: the setup script includes base `migrate/collectstatic` commands, as well as database seeding. It's that seeding that's now idempotent)

### CODE REVIEW

File: infra/README.md
Comment (grayside) [inline infra/README.md:47]: question: Some of these changes seem like unrelated readme docs regeneration. Should there be a separate process to make that happen?

File: infra/postdeployment.tf
Comment (grayside) [inline infra/postdeployment.tf:85]: future: I realize it's a bit out of scope for the PR, but looking at this startup script, it took me a double-take to realize we probably don't want to run with `set -e` because we want all these things to happen, they don't depend on each other. It might be worth adding a comment to explain.

File: infra/postdeployment.tf
Comment (grayside) [inline infra/postdeployment.tf:48]: question: Confirming my understanding, without this `desired_status`, terraform would look, see we have an instance, and no-op. However, with this config in place, terraform is instructed to start up the GCE instance if it's not running.

File: infra/postdeployment.tf
Comment (grayside) [inline infra/postdeployment.tf:46]: question: Is there any particular reason why not use [cloud_run_v2_job](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloud_run_v2_job) or [google_cloudbuild_trigger](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloudbuild_trigger) instead of GCE?

File: infra/postdeployment.tf
Comment (grayside) [inline infra/postdeployment.tf:85]: question: How do we know if something goes wrong with the startup. For example, what if the migration job stops working? Does something alert, or is this a place we should consider adding a log-based alert?

File: infra/variables.tf
Comment (grayside) [inline infra/variables.tf:58]: thought: Normally I might suggest we update Avocano in a separate PR, but I recognize this is carrying a change required for this PR to work.

**Review by grayside — APPROVED** [review]
I raised a couple non-blocking out-of scope points, and some questions that are more about my learning the project than any challenge to this change. I've prefixed review comments based on https://conventionalcomments.org/ to help make the context of each point explicit.

LGTM and merge when ready.

File: infra/README.md
Comment (glasnt) [reply infra/README.md:47]: Looking into this more: Normally yes, but there was an issue with the linter that was fixed in https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/pull/1484

https://console.cloud.google.com/cloud-build/builds/94a0ce8b-8629-41bc-be5c-1953d7d9a164?project=2843445864 (from earlier in this PR) successfully showed it failed.

https://console.cloud.google.com/cloud-build/builds/1a81afbd-3fbc-42f8-befe-6fbbd027494b?project=2843445864 (from #37) failed, but the error code was incorrect so the status was 'success'. 

Since the linting was failing on my PR, I fixed them. I could split them out into a different PR, but then this PR would continue to fail until that PR was merged and this rebased.

File: infra/postdeployment.tf
Comment (glasnt) [reply infra/postdeployment.tf:48]: Correct. 

Terraform takes a convergence style approach: it does the minimum set of steps required to bring the infra back into the state documented. For the aspects of the GCE state described, that would reflect reality, so no changes would be made. Now that we're adding "Oh it also needs to be running", when we're elsewhere making sure it shuts down, TF will always assert that state, which has the side-benefit of running the metadata script (see later comment for why).

File: infra/postdeployment.tf
Comment (glasnt) [reply infra/postdeployment.tf:46]: We are using cloud_run_v2_job, see `infra/jobs.tf`, but while TF can ensure these resources exist, TF cannot execute them. The current solution to this issue is the GCE metadata startup script workaround. 

Cloud Build would just either call the jobs themselves (what the metadata startup script currently encapsulates), or break out the jobs into literal Cloud Build (but since we want them separate later for tutorials, this was the simplest solution). Avocano's self-contained TF opts to use Cloud Build.

File: infra/postdeployment.tf
Comment (glasnt) [reply infra/postdeployment.tf:85]: I'm going to separate this and the other comment on this line into a Feature Request for later processing. Since metadata startup script runs async to TF, it can't block TF, so we'd need to have a separate process for that.

Comment (glasnt) [reply infra/postdeployment.tf:85]: https://github.com/GoogleCloudPlatform/terraform-dynamic-python-webapp/issues/50

Comment (glasnt) [reply infra/postdeployment.tf:85]: https://github.com/GoogleCloudPlatform/terraform-dynamic-python-webapp/issues/50

**Pull request merged**
